content/intro-to-storybook/react/ja/test.md

```diff
@@ -0,0 +1,121 @@
+---
+title: 'UI コンポーネントをテストする'
+tocTitle: 'テスト'
+description: 'UI コンポーネントのテスト手法について学びましょう'
+commit: '3e283f7'
+---
+
+Storybook のチュートリアルをテスト抜きには終われません。テストは高品質な UI を作成するのに必要なことです。疎結合なシステムにおいては、些細な変更で大きなリグレッションをもたらすことがあるのです。既に 3 種類のテストについて学びました:
+
+- **手動テスト** コンポーネントの正しさを開発者が手動で確認します。コンポーネントを作成する際、見た目が問題ないかチェックするのに役立ちます。
+- **スナップショットテスト** Storyshots を使用し、コンポーネントのマークアップを記録します。描画時のエラーや警告の原因となるマークアップの変更を常に把握するのに役立ちます。
+- **単体テスト** Jest を使用し、コンポーネントへの決まった入力より同様の出力が出ていることを確認します。コンポーネントの機能性をテストするのに優れています。
+
+## 「でも、見た目は大丈夫？」
+
+残念ながら、前述のテスト方法だけでは UI のバグを防ぎきれません。UI のデザインというのは主観的で、ごくわずかな変更で全く異なってしまうことがあるため、テストが厄介なのです。手動テストは、その名の通り、手動です。UI のスナップショットテストでは多数の偽陽性を発生させてしまいます。ピクセルレベルの単体テストは価値があまりありません。Storybook のテスト戦略にはビジュアルテストが不可欠です。
+
+## Storybook 向けのビジュアルテスト
+
+視覚的なリグレッションテスト (ビジュアルテスト) は、見た目の変更を検出するために設計されています。ビジュアルテストはコミット毎に各ストーリーのスクリーンショットを撮って、前のコミットと比較して変更点を探します。レイアウトや色、サイズ、コントラストといった表示要素の確認にとても適しています。
+
+<video autoPlay muted playsInline loop style="width:480px; margin: 0 auto;">
+  <source
+    src="/intro-to-storybook/visual-regression-testing.mp4"
+    type="video/mp4"
+  />
+</video>
+
+Storybook は視覚的なリグレッションテスト用の素晴らしいツールです。Storybook において、すべてのストーリーはテスト仕様となるからです。ストーリーを書くたび、仕様が無料でついてきます！
+
+視覚的なリグレッションテスト向けのツールは多々あります。Storybook のメンテナーが作成した無料の発行サービスである [**Chromatic**](https://www.chromatic.com/) がおすすめです。Chromatic はクラウド上でビジュアルテストを並列実行します。[前の章](/react/ja/deploy/)と同じように Storybook をインターネット上に発行出来ます。
+
+## UI の変更を検知する
+
+視覚的なリグレッションテストでは UI コードにより描画されたイメージが基準となるイメージと比較されます。ツールが UI の変更が検知したら、教えてくれます。
+
+それでは、`Task` コンポーネントの背景を変更し、どう動くのか見てみましょう。
+
+変更する前に新しいブランチを作成します:
+
+```bash
+git checkout -b change-task-background
+```
+
+`Task` を以下のように変更します:
+
+```js
+// src/components/Task.js
+<div className="title">
+  <input
+    type="text"
+    value={title}
+    readOnly={true}
+    placeholder="Input title"
+    style={{ textOverflow: 'ellipsis', background: 'red' }}
+  />
+</div>
+```
+
+これでタスクの背景色が変更されます。
+
+![task background change](/intro-to-storybook/chromatic-task-change.png)
+
+この変更をステージングします:
+
+```bash
+git add src\components\Task.js
+```
```

Comment:
@chameleonhead when you're able please adjust the snippet to "git add .". With a recent change that snippet was changed and it might cause some confusion to non windows users and to avoid that, we probably can make a bit more "agnostic" and go with i'm proposing.